### PR TITLE
oss-perf: modify default protocol for ossfs URL

### PR DESCRIPTION
**Pull request by AlbeeSo**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
In current version, CSI will not check protocol of ossfs URL. Then, ossfs will use default protocol (http) if it is not specified explicitly. 
In this PR, protocol will be set by CSI. If instance needs to access OSS with public endpoint, use https as default protocol instead.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/oss/utils.go [hunk 1]
```diff
@@ -126,3 +126,18 @@ func doMount(mounter mountutils.Interface, target string, opts Options, mountOpt
 	logger.Info("mounted successfully")
 	return nil
 }
+
+func modifiedURL(originURL, regionID string) (URL string, modified bool) {
```
Comment (mowangdk) [inline pkg/oss/utils.go:130]: Let's split this method into  ```setNetworkType``` and ```setTransmissionProtocol```

Comment (mowangdk) [reply pkg/oss/utils.go:130]: and we should add unit test for these two method

File: pkg/oss/utils.go [hunk 2]
```diff
@@ -126,3 +126,23 @@ func doMount(mounter mountutils.Interface, target string, opts Options, mountOpt
...
+	}
+	return
+}
+
+func setTransmissionProtocol(originURL string) (URL string, modified bool) {
+	URL = originURL
+	if strings.HasPrefix(URL, "http://") || strings.HasPrefix(URL, "https://") {
+		return
+	}
+	if strings.HasSuffix(URL, "-internal.aliyuncs.com") {
```
Comment (huww98) [inline pkg/oss/utils.go:144]: I think at least URL like `oss-cn-beijing-internal.aliyuncs.com/` is vaild. Paths other than '/' should be considered too. scheme and hostname parts of URL are case insensitive, and should be considered.

I think we may parse the URL with `net/url` package.

**Review by huww98 — CHANGES_REQUESTED** [review]

File: pkg/oss/utils.go (hunk 2 above)
Comment (AlbeeSo) [reply pkg/oss/utils.go:144]: [url here is can only ](https://help.aliyun.com/zh/oss/user-guide/regions-and-endpoints?spm=a2c4g.11186623.0.0.1e2030f0DijQDq#section-oao-7ao-11f)
Endpoint not matches  "(http://,https://)oss-{{region}}-{{pub/internal/null}}-aliyuncs.com" is invalid to access OSS by ossfs. Use the raw URL or the modified one will both cause an error.

Comment (huww98) [reply pkg/oss/utils.go:144]: I don't get it. The provided link shows endpoint, which is only the hostname part of URL. URL is a term defined by [rfc3986](https://datatracker.ietf.org/doc/html/rfc3986). Case insensitivity of scheme should be supported:
> Although schemes are case-
   insensitive, the canonical form is lowercase and documents that
   specify schemes must do so with lowercase letters.  An implementation
   should accept uppercase letters as equivalent to lowercase in scheme
   names (e.g., allow "HTTP" as well as "http") for the sake of
   robustness but should only produce lowercase scheme names for
   consistency.

DNS name should also be case insensitive (see [rfc4343](https://datatracker.ietf.org/doc/html/rfc4343)).

As for the path component. I've verified that `oss-cn-beijing-internal.aliyuncs.com/` this URL works currently. We may also want to support this too.

Comment (AlbeeSo) [reply pkg/oss/utils.go]: In fact, value of -ourl= for s3fs/ossfs is little bit different from a real URL. It's for the Endpoint part of Host, which is equal to BucketName.Endpoint. So 'URL' here can only be endpoint or the endpoint with protocol.  I fixed case like `oss-cn-beijing-internal.aliyuncs.com/` . 
https://help.aliyun.com/zh/oss/developer-reference/common-http-headers?spm=a2c4g.11186623.0.0.3b2b3864sI0iwV#section-eq1-b2y-wdb

**Pull request merged**